pkg/registry/core/service/ipallocator/allocator.go

```diff
@@ -165,31 +167,58 @@ func (r *Range) CIDR() net.IPNet {
 // or has already been reserved.  ErrFull will be returned if there
 // are no addresses left.
 func (r *Range) Allocate(ip net.IP) error {
+	label := r.CIDR()
 	ok, offset := r.contains(ip)
 	if !ok {
+		// update metrics
```

Comment:
are we tracking user error?

Reply:
the metrics is `allocation_errors_total` , I didn't want to discriminate on the error type.
I don't expect users to use this metric to narrow down problems, they should go to the apiserver audit.log for that, instead, I expect this to be a measure of the allocation process health, as in "hey, you are having error allocating ips, you should take a look on what is going on here"